switchPairs.rb

```diff
@@ -0,0 +1,37 @@
+# def switchPairs(a)
+  # can swap with parallel assignment - I have a hunch this might be a ruby specific method, since we talked about how Ruby arrays are special
+  # a[0], a[1] = a[1], a[0]
+  # a[2], a[3] = a[3], a[2]
+  # a[4], a[5] = a[5], a[4]
+
+#   (0..a.length-1).each do |i|
+#     if i % 2 == 0
+#       a[i], a[i+1] = a[i+1], a[i]
+#     end
+#   end
+#   return a
+# end
+
+def switchPairs(a)
+  if a.length % 2 == 0
+    (0..a.length-1).each do |i|
+      if i % 2 == 0
+        a[i], a[i+1] = a[i+1], a[i]
+      end
+    end
+  else
+    (0..a.length-2).each do |i|
+      if i % 2 == 0
+        a[i], a[i+1] = a[i+1], a[i]
+      end
+    end
+  end
+  return a
+end
```

Comment:
There are some ways that you could have shortened up this code without having to handle odd/even cases separately, but looks like it works!